code-for-canada/project-thundercat
# [Registration Form] Adding new fields (DOB, PRI / Military # / Privacy Notice)

# Description

This PR is introducing the new registration form fields, added in the new version of the design. The following fields have been added:

- Date of birth (DD - MM ---Y)
- PRI or Military number
- Privacy Notice section (check mark + description)

All validation have also been added for each of these new fields.

## Type of change

- New feature (non-breaking change which adds functionality)

## Screenshot

**Initial Form:**

![image](https://user-images.githubusercontent.com/23021242/59055472-e5336000-8863-11e9-8c78-877d033d8ea0.png)

**Validation errors related to the added fields:**

![image](https://user-images.githubusercontent.com/23021242/59055546-0f851d80-8864-11e9-8747-f6c7d034fceb.png)

**When all added fields are valid:**

![image](https://user-images.githubusercontent.com/23021242/59055578-2deb1900-8864-11e9-8cfc-75d958eff35d.png)

**Privacy Notice Popup (TODO):**

![image](https://user-images.githubusercontent.com/23021242/59055606-3e9b8f00-8864-11e9-9a2c-bb75867e4330.png)

# Testing

Manual steps to reproduce this functionality:

1.  Open the registration form from the home page.
2.  Fill out the form by adding voluntary errors (numbers in name fields, special character in DOB, etc.) and make sure that everything is working as expected.

# Checklist

Applicable for all code changes.
